File: implementation/scipamato/common/entity/src/test/java/ch/difty/scipamato/common/entity/ScipamatoEntityTest.java
package ch.difty.scipamato.common.entity;

import static ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.CREATED;
import static ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.MODIFIED;
import static org.assertj.core.api.Assertions.assertThat;

import java.time.LocalDateTime;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class ScipamatoEntityTest {

    private static final int VERSION = 10;

    private static final LocalDateTime CD = LocalDateTime
        .now()
        .minusDays(1);
    private static final LocalDateTime LM = LocalDateTime
        .now()
        .plusDays(1);

    private final ScipamatoEntity e = new ScipamatoEntity();

    @BeforeEach
    void setUp() {
        e.setCreated(CD);
        e.setLastModified(LM);
        e.setVersion(VERSION);
    }

    @Test
    void get() {
        assertThat(e.getCreated()).isEqualTo(CD);
        assertThat(e.getLastModified()).isEqualTo(LM);
        assertThat(e.getVersion()).isEqualTo(VERSION);
    }

    @Test
    void testingToString() {
        assertThat(e.toString()).isEqualTo(
            "ScipamatoEntity[created=" + CD + ",lastModified=" + LM + ",version=" + VERSION + "]");
    }

    @Test
    void equals() {
        EqualsVerifier
            .forClass(ScipamatoEntity.class)
            .withIgnoredFields(CREATED.getName(), MODIFIED.getName())
            .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
            .verify();
    }

}

File: implementation/scipamato/common/entity/src/test/kotlin/ch/difty/scipamato/common/entity/ScipamatoEntityTest.kt
package ch.difty.scipamato.common.entity

import ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.CREATED
import ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.MODIFIED
import nl.jqno.equalsverifier.EqualsVerifier
import nl.jqno.equalsverifier.Warning
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.time.LocalDateTime

class ScipamatoEntityTest {

    private val e = ScipamatoEntity()

    @BeforeEach
    fun setUp() {
        e.created = CD
        e.lastModified = LM
        e.version = VERSION
    }

    @Test
    fun get() {
        assertThat(e.created).isEqualTo(CD)
        assertThat(e.lastModified).isEqualTo(LM)
        assertThat(e.version).isEqualTo(VERSION)
    }

    @Test
    fun testingToString() {
        assertThat(e.toString()).isEqualTo("ScipamatoEntity[created=$CD,lastModified=$LM,version=$VERSION]")
    }

    @Test
    fun equals() {
        EqualsVerifier
                .forClass(ScipamatoEntity::class.java)
                .usingGetClass()
                .withIgnoredFields(CREATED.fieldName, MODIFIED.fieldName)
                .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
                .verify()
    }

    companion object {
        private const val VERSION = 10

        private val CD = LocalDateTime.now().minusDays(1)
        private val LM = LocalDateTime.now().plusDays(1)
    }

}
